crates/conformance/src/bin/spin-conformance.rs

```diff
@@ -0,0 +1,58 @@
+use anyhow::Result;
+use clap::Parser;
+use std::{
+    fs::{self, File},
+    io::{self, Read, Write},
+    path::PathBuf,
+};
+use wasmtime::{Config, Engine, Module};
+
+#[derive(Parser)]
+#[clap(author, version, about)]
+pub struct Options {
+    /// Name of Wasm file to test (or stdin if not specified)
+    #[clap(short, long)]
+    pub input: Option<PathBuf>,
+
+    /// Name of JSON file to write report to (or stdout if not specified)
+    #[clap(short, long)]
+    pub output: Option<PathBuf>,
+
+    /// Name of TOML configuration file to use
+    #[clap(short, long)]
+    pub config: Option<PathBuf>,
+}
+
+fn main() -> Result<()> {
+    let options = &Options::parse();
+
+    let engine = &Engine::new(&Config::new())?;
+
+    let module = &if let Some(input) = &options.input {
+        Module::from_file(engine, input)
+    } else {
+        Module::new(engine, &{
+            let mut buffer = Vec::new();
+            io::stdin().lock().read_to_end(&mut buffer)?;
```

Comment:
Yeah, I guess I've gotten in the habit of explicitly locking since it can help with performance e.g. when passing it as a `Read` or `Write` to another method and I want to lock just once instead of on every read or write.  Won't make a difference in this case, though.

Reply:
I've removed the `lock` call on that line but kept it in the stdout case to make it clear that we want to write the JSON atomically.